docs/extensibility/snippets/MailDevResource/MailKit.Client/MailKitClientFactory.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Net;
+using MailKit.Net.Smtp;
+
+namespace MailKit.Client;
+
+/// <summary>
+/// A factory for creating <see cref="ISmtpClient"/> instances
+/// given a <paramref name="smtpUri"/> (and optional <paramref name="credentials"/>).
+/// </summary>
+/// <param name="settings">
+/// The <see cref="MailKitClientSettings"/> settings for the SMTP server
+/// </param>
+/// <param name="credentials">
+/// The optional <see cref="ICredentials"/> used to authenticate to the SMTP server
+/// </param>
+public sealed class MailKitClientFactory(
+    MailKitClientSettings settings,
+    ICredentials? credentials = null) : IDisposable
```

Comment:
How does `credentials` ever get set?

```suggestion
public sealed class MailKitClientFactory(MailKitClientSettings settings) : IDisposable
```

We should be reading the credentials from `settings`.